sormas-app/app/src/main/java/de/symeda/sormas/app/util/DiseaseConfigurationCache.java

```diff
@@ -76,9 +70,12 @@ private DiseaseConfigurationCache() {
 			} else {
 				nonPrimaryDiseases.add(disease);
 			}
-			if (Boolean.TRUE.equals(configuration.getCaseBased()) || (configuration.getCaseBased() == null && disease.isDefaultCaseBased())) {
+			if (Boolean.TRUE.equals(configuration.getCaseSurveillanceEnabled())
+				|| (configuration.getCaseSurveillanceEnabled() == null && disease.isDefaultCaseBased())) {
 				caseBasedDiseases.add(disease);
-			} else {
+			}
+			if (Boolean.TRUE.equals(configuration.getAggregateReportingEnabled())
+				|| (configuration.getAggregateReportingEnabled() == null && !disease.isDefaultCaseBased())) {
```

Comment:
Diseases should also get a `isDefaultAggregateReportingEnabled` method which should be used here, because this logic is flawed (i.e. it won't work if a disease is enabled for both case surveillance and aggregate reporting by default). All diseases that so far have been neither primary nor case based should be enabled for aggregate reporting by default.

Reply:
logic updated